tests/src/item.test.ts

```diff
@@ -0,0 +1,398 @@
+            inherited: { notInherited: true },
+          },
+        ],
+        buildPermissiveness: [
+          {
+            permissivenessType: { tokenHolder: true },
+            inherited: { notInherited: true },
+          },
+        ],
+        stakingWarmUpDuration: null,
+        stakingCooldownDuration: null,
+        stakingPermissiveness: null,
+        unstakingPermissiveness: null,
+        childUpdatePropagationPermissiveness: [
+          {
+            childUpdatePropagationPermissivenessType: { usages: true },
+            inherited: { notInherited: true },
+          },
+          {
+            childUpdatePropagationPermissivenessType: { components: true },
+            inherited: { notInherited: true },
+          },
+          {
+            childUpdatePropagationPermissivenessType: {
+              updatePermissiveness: true,
+            },
+            inherited: { notInherited: true },
+          },
+          {
+            childUpdatePropagationPermissivenessType: {
+              buildPermissiveness: true,
+            },
+            inherited: { notInherited: true },
+          },
+          {
+            childUpdatePropagationPermissivenessType: {
+              stakingPermissiveness: true,
+            },
+            inherited: { notInherited: true },
+          },
+          {
+            childUpdatePropagationPermissivenessType: {
+              freeBuildPermissiveness: true,
+            },
+            inherited: { notInherited: true },
+          },
+        ],
+      },
+      config: {
+        usageRoot: null,
+        usageStateRoot: null,
+        componentRoot: null,
+        usages: null,
+        components: null,
+      },
+    };
+
+    args = {
+      classIndex: new anchor.BN(itemClassIndex),
+      parentClassIndex: null,
+      space: new anchor.BN(300),
+      desiredNamespaceArraySize: 1,
+      updatePermissivenessToUse: { tokenHolder: true },
+      storeMint: false,
+      storeMetadataFields: false,
+      itemClassData,
+      parentOfParentClassIndex: null,
+    } as Instructions.Item.CreateItemClassArgs;
+
+    accounts = {
+      itemMint: itemClassMintKeypair.publicKey,
+      parent: null,
+      parentMint: null,
+      parentOfParentClassMint: null,
+      metadataUpdateAuthority: walletKeypair.publicKey,
+      parentUpdateAuthority: null,
+    } as Instructions.Item.CreateItemClassAccounts;
+
+    additionalArgs = {} as Instructions.Item.CreateItemClassAdditionalArgs;
+
+    const createItemClassIxs = await itemProgram.instruction.createItemClass(
+      args,
+      accounts,
+      additionalArgs
+    );
+
+    ixs.push(...createItemClassMintIxs, ...createItemClassIxs);
+
+    const { txid: createItemClassTxid } =
+      await Transaction.sendTransactionWithRetry(
+        provider.connection,
+        provider.wallet,
+        ixs,
+        [itemClassMintKeypair]
+      );
+
+    expect(createItemClassTxid).toBeDefined();
```

Comment:
It would be nice to assert the fields that were set in the ItemClass are indeed set as we think

Reply:
Yeah, we need better assertion. Can I improve it after the merge?

Reply:
I think we should just take care of it now, otherwise I dont see a lot of value in the tests if we arent checking the fields

Reply:
Yeah if you want to do them later then you should rename this pr to "add system-wide test structure". We would then immediately pickup adding the actual test cases. You should also mark these tests as pending/skipped until they are fixed.